main.rb

```diff
@@ -3,89 +3,64 @@
 require_relative 'student'
 require_relative 'teacher'
 require_relative 'app'
+require_relative 'logics'
 
-def print_menu
-  puts ''
-  puts 'Welcome To School Library App 🤩'
-  puts '*************************'
-  puts ''
-  puts 'Please Enter An Option by Entering a number'
-  puts '1 - List all books.'
-  puts '2 - List all people.'
-  puts '3 - Create a person (teacher or student, not a plain Person).'
-  puts '4 - Create a book.'
-  puts '5 - Create a rental.'
-  puts '6 - List all rentals for a given person id.'
-  puts '7 - Exit ❌'
-  puts ''
-end
-
-def process_list_books(app, books)
-  app.list_books(books)
-end
-
-def process_list_people(app, people)
-  app.list_people(people)
-end
+class Main < Logics
+  def process_list_books(app, books)
+    app.list_books(books)
+  end
 
-def create_person(app, people)
-  puts 'Do you want to create a student (1) or a teacher (2)? [Input Number (1) or (2)]:'
-  type = gets.chomp.to_i
-  case type
-  when 1
-    app.create_student(people)
-  when 2
-    app.create_teacher(people)
-  else
-    puts 'Invalid option'
+  def process_list_people(app, people)
+    app.list_people(people)
   end
-end
 
-def process_add_book(app, books)
-  app.add_book(books)
-end
+  def process_add_book(app, books)
+    app.add_book(books)
+  end
 
-def process_create_rental(app, books, people, rentals)
-  app.create_rental(books, people, rentals)
-end
+  def process_create_rental(app, books, people, rentals)
+    app.create_rental(books, people, rentals)
+  end
 
-def process_list_rentals(app, rentals)
-  app.list_rentals(rentals)
-end
+  def process_list_rentals(app, rentals)
+    app.list_rentals(rentals)
+  end
```

Comment:
- [ ] Also, If you notice, most of these methods are just wrappers over the ones you have in your `app.rb` file. When you apply the changes I suggested, you shouldn't need any of these, just the ones in your app, ending up with less code and simpler logic KISS! 🤓 🔧